### ISSUE TITLE
`export default interface` in augmentation shadows original default export

### ISSUE THREAD

**Issue by mattmccutchen** [microsoft/TypeScript#27286]
**TypeScript Version:**  master (5fb3976)

**Search Terms:** default export interface augmentation 

**Code**

```ts
declare module "example" {
    const y = 42;
    export default y;
    
    // Augmentation
    module "example" {
        export default interface Foo {}
    }
}

declare module "example2" {
    import y from "example";
    // Actual: error TS2693: 'y' only refers to a type, but is being used as a value here.
    const yy: typeof y;
}
```

**Expected behavior:** No error (declarations merge properly), or error on `export default interface Foo {}` that default exports in augmentations aren't supported.

**Actual behavior:** Error as marked.

**Playground Link:** [Link](https://www.typescriptlang.org/play/#src=declare%20module%20%22example%22%20%7B%0D%0A%20%20%20%20const%20y%20%3D%2042%3B%0D%0A%20%20%20%20export%20default%20y%3B%0D%0A%20%20%20%20%0D%0A%20%20%20%20%2F%2F%20Augmentation%0D%0A%20%20%20%20module%20%22example%22%20%7B%0D%0A%20%20%20%20%20%20%20%20export%20default%20interface%20Foo%20%7B%7D%0D%0A%20%20%20%20%7D%0D%0A%7D%0D%0A%0D%0Adeclare%20module%20%22example2%22%20%7B%0D%0A%20%20%20%20import%20y%20from%20%22example%22%3B%0D%0A%20%20%20%20%2F%2F%20Actual%3A%20error%20TS2693%3A%20'y'%20only%20refers%20to%20a%20type%2C%20but%20is%20being%20used%20as%20a%20value%20here.%0D%0A%20%20%20%20const%20yy%3A%20typeof%20y%3B%0D%0A%7D%0D%0A)

**Related Issues:** #14080

**Comment by RyanCavanaugh** [comment]
No longer an error
